src/server/scripts/Outland/zone_hellfire_peninsula.cpp

```diff
@@ -420,10 +420,488 @@ class npc_fel_guard_hound : public CreatureScript
     }
 };
 
+enum ExorcismSpells
+{
+    SPELL_JULES_GOES_PRONE     = 39283,
+    SPELL_JULES_THREATENS_AURA = 39284,
+    SPELL_JULES_GOES_UPRIGHT   = 39294,
+    SPELL_JULES_VOMITS_AURA    = 39295,
+
+    SPELL_BARADAS_COMMAND      = 39277,
+    SPELL_BARADA_FALTERS       = 39278,
+};
+
+enum ExorcismTexts
+{
+    SAY_BARADA_1 = 0,
+    SAY_BARADA_2 = 1,
+    SAY_BARADA_3 = 2,
+    SAY_BARADA_4 = 3,
+    SAY_BARADA_5 = 4,
+    SAY_BARADA_6 = 5,
+    SAY_BARADA_7 = 6,
+    SAY_BARADA_8 = 7,
+
+    SAY_JULES_1  = 0,
+    SAY_JULES_2  = 1,
+    SAY_JULES_3  = 2,
+    SAY_JULES_4  = 3,
+    SAY_JULES_5  = 4,
+};
+
+Position const exorcismPos[11] =
+{
+    { -707.123f, 2751.686f, 101.592f, 4.577416f }, //Barada Waypoint-1      0
+    { -710.731f, 2749.075f, 101.592f, 1.513286f }, //Barada Cast position   1
+    { -710.332f, 2754.394f, 102.948f, 3.207566f }, //Jules                  2
+    { -714.261f, 2747.754f, 103.391f, 0.0f },      //Jules Waypoint-1       3
+    { -713.113f, 2750.194f, 103.391f, 0.0f },      //Jules Waypoint-2       4
+    { -710.385f, 2750.896f, 103.391f, 0.0f },      //Jules Waypoint-3       5
+    { -708.309f, 2750.062f, 103.391f, 0.0f },      //Jules Waypoint-4       6
+    { -707.401f, 2747.696f, 103.391f, 0.0f },      //Jules Waypoint-5       7
+    { -708.591f, 2745.266f, 103.391f, 0.0f },      //Jules Waypoint-6       8
+    { -710.597f, 2744.035f, 103.391f, 0.0f },      //Jules Waypoint-7       9
+    { -713.089f, 2745.302f, 103.391f, 0.0f },      //Jules Waypoint-8      10
+};
+
+enum ExorcismMisc
+{
+    NPC_DARKNESS_RELEASED               = 22507,
+    NPC_FOUL_PURGE                      = 22506,
+    NPC_COLONEL_JULES                   = 22432,
+
+    BARADAS_GOSSIP_MESSAGE              = 10683,
+
+    QUEST_THE_EXORCISM_OF_COLONEL_JULES = 10935,
+
+    ACTION_START_EVENT                  = 1,
+    ACTION_JULES_HOVER                  = 2,
+    ACTION_JULES_FLIGHT                 = 3,
+    ACTION_JULES_MOVE_HOME              = 4,
+};
+
+enum ExorcismEvents
+{
+    EVENT_BARADAS_TALK = 1,
+
+    //Colonel Jules
+    EVENT_SUMMON_SKULL = 1,
+};
+
+/*######
+## npc_barada
+######*/
+
+class npc_barada : public CreatureScript
+{
+public:
+    npc_barada() : CreatureScript("npc_barada") { }
+       
+    struct npc_baradaAI : public ScriptedAI
+    {
+        npc_baradaAI(Creature* creature) : ScriptedAI(creature)
+        {
+            Initialize();
+        }
+
+        void Initialize()
+        {
+            step = 0;
+        }
+
+        void Reset() override
+        {
+            events.Reset();
+            step = 0;
```

Comment:
I thought `step` is already initialized ? Then again, I don't remember how it should look.